drivers/net/wireless/mediatek/mt76/mt7921/mcu.c

```diff
@@ -1118,11 +1118,7 @@ int mt7921_mcu_set_chan_info(struct mt7921_phy *phy, int cmd)
 		req.center_ch2 = ieee80211_frequency_to_channel(freq2);
 	}
 
-	err = mt76_mcu_send_msg(&dev->mt76, cmd, &req, sizeof(req), true);
-	if (err < 0)
-		return err;
-
-	return mt76_connac_mcu_set_rate_txpower(phy->mt76);
+	return mt76_mcu_send_msg(&dev->mt76, cmd, &req, sizeof(req), true);
```

Comment:
@objelf this does seems correct to me since eeprom code will parse dt rate-power data according to the operating channel. Doing so we can use the values from a wrong channel since the regdomain can be set w/o changing the channel.

Reply:
Firmware would manage all channels (e.g hardware scan and network concurrency among multiple clients) so firmware has to know all tx power for all rates on all channels at one time. 
That is SDK doing so.